src/components/smart/Bluetooth.js

```diff
@@ -71,51 +101,69 @@ export default class Bluetooth extends Component {
     }
 
     /**
-     * TODO: (1 pts)
-     *      Remove the three handlers added in the componentDidMount (3)
+     * TODO: (1 pts) x
+     *      Remove the three handlers added in the componentDidMount (3) x
      */
     componentWillUnmount() {
         /// code expected
+        this.handleDiscoverPeripheral.remove();
+        this.handleStopScan.remove();
+        this.handleDisconnectedPeripheral.remove();
+        this.handleAppStateChange.remove();
+
         console.log("Bluetooth: Component will unmount not implemented");
     }
 
     handleDisconnectedPeripheral(data) {
-        console.log('Disconnected from ' + data.peripheral);
+        console.log('Disconnected from ' + data.peripherals);
     }
 
     /**
-     * TODO: (1 pts)
-     *      Set "scanning" to false
+     * TODO: (1 pts) x
+     *      Set "scanning" to false x
      */
     handleStopScan() {
         /// code expected
-        console.log("Bluetooth: Handle stop scan not implemented");
+        console.log('Scan is stopped');
+         this.setState({ scanning: false });
     }
 
     /**
      * TODO: (4 pts)
-     *  (1) If already scanning, do nothing
-     *  (2) Start the scan for a duration of 10 seconds
+     *  (1) If already scanning, do nothing x
+     *  (2) Start the scan for a duration of 10 seconds x
      *  (3) Search and memorize devices that have the Battery Level Service
      *  (4) Correctly handle possible exceptions
      */
     startScan() {
         /// code expected
+        if (!this.state.scanning) {
+              this.setState({peripherals: new Map()});
+              BleManager.scan([], 10, true).then((results) => {
+                console.log('Scanning...');
+                this.setState({scanning:true});
+              });
+        }
         console.log("Bluetooth: Start scan not implemented");
 
     }
 
     /**
      * TODO: (3 pts)
      *
-     *  (1) Check to see if the new peripheral already exists in peripherals
-     *  (2) If it doesn't exist then add the new peripheral to the peripherals Map object
-     *  (3) Update the state with peripherals
+     *  (1) Check to see if the new peripheral already exists in peripherals x
+     *  (2) If it doesn't exist then add the new peripheral to the peripherals Map object x
+     *  (3) Update the state with peripherals x
      *
      * @param peripheral
      */
     handleDiscoverPeripheral(peripheral){
         /// code expected
+        peripherals: new Map()
```

Comment:
This will cause an error and if it does work then it'll just delete everything. 2/3